src/main/java/com/nike/cerberus/operation/core/CreateCompleteCerberusDataS3BackupOperation.java

```diff
@@ -0,0 +1,395 @@
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    private static final String AWS_PROVIDER = "AWS";
+
+    private final ObjectMapper objectMapper;
+    private final ConfigStore configStore;
+    private final CerberusAdminClient cerberusAdminClient;
+    private final MetricsService metricsService;
+    private final EnvironmentMetadata environmentMetadata;
+    private final AWSSecurityTokenService sts;
+    private final Map<String, S3StoreService> regionToEncryptedStoreServiceMap = new HashMap<>();
+
+    @Inject
+    public CreateCompleteCerberusDataS3BackupOperation(CerberusAdminClientFactory cerberusAdminClientFactory,
+                                                       @Named(CerberusModule.CONFIG_OBJECT_MAPPER)
+                                                               ObjectMapper objectMapper,
+                                                       ConfigStore configStore,
+                                                       MetricsService metricsService,
+                                                       EnvironmentMetadata environmentMetadata,
+                                                       AWSSecurityTokenService sts) {
+
+        this.objectMapper = objectMapper;
+        this.configStore = configStore;
+        this.metricsService = metricsService;
+        this.environmentMetadata = environmentMetadata;
+        this.sts = sts;
+
+        cerberusAdminClient = cerberusAdminClientFactory.getNewCerberusAdminClient(
+                configStore.getCerberusBaseUrl());
+    }
+
+    @Override
+    public void run(CreateCompleteCerberusDataS3BackupCommand command) {
+        log.info("Starting Cerberus Backup");
+
+        List<String> regionsToStoreBackups = command.getBackupRegions();
+        validateRegions(regionsToStoreBackups);
+
+        Date now = new Date();
+        String prefix = new SimpleDateFormat("yyyy-MM-dd-HH-mm-ss").format(now);
+        List<SafeDepositBox> sdbMetadataList = cerberusAdminClient.getAllSdbMetadata();
+
+        int sdbCount = sdbMetadataList.size();
+        if (sdbCount < 1) {
+            throw new RuntimeException("CMS returned no data when listing SDB metadata");
+        } else {
+            log.info(String.format("CMS returned that there are %s SDBs to back up.", sdbMetadataList.size()));
+        }
+
+        CerberusSdbMetadata cerberusSdbMetadata = new CerberusSdbMetadata();
+        for (SafeDepositBox sdb : sdbMetadataList) {
+            log.info(String.format("Backing up %s", sdb.getName()));
+            Map<String, Map<String, String>> vaultData = recurseVault(sdb.getPath(), new HashMap<>());
+            sdb.setData(vaultData);
+            String key = sdb.getName().toLowerCase().replaceAll("/\\W+/", "-");
+            saveDataToS3(sdb, prefix, key, regionsToStoreBackups);
+            cerberusSdbMetadata = processMetadata(sdb, cerberusSdbMetadata);
+        }
```

Comment:
I don't want the try catch because then we might not know that a backup is failing, as it is right now the command will exit with status 1 if anything is not right.

As for the paths, you already get that with --debug, so I don't think its worth adding it again.